lib_expense/src/main/java/com/example/expenselogger/lib_expense/data/impl/ExpenseContractImpl.kt

```diff
@@ -0,0 +1,27 @@
+package com.example.expenselogger.lib_expense.data.impl
+
+import com.example.expenselogger.cache.repository.ExpenseRepository
+import com.example.expenselogger.lib_expense.data.mapper.ExpenseModelMapper
+import com.example.expenselogger.lib_expense.domain.contract.ExpenseContract
+import com.example.expenselogger.lib_expense.domain.model.Expense
+import javax.inject.Inject
+
+internal class ExpenseContractImpl @Inject constructor(
+    private val expenseRepository: ExpenseRepository,
+    private val expenseModelMapper: ExpenseModelMapper
+) : ExpenseContract {
+    override suspend fun insertExpense(expense: Expense): Long =
```

Comment:
What will the id returned in the insert method be used for

Reply:
Ohhhh... there's nothing I can think of, except for the contract test

Reply:
Yeah if you can't see where it's going to be used then there's no need having it here. Think about the entire system at a high level. What happens when you insert an item into the db, what should be the effects of that operation. It'll guide how you structure the classes

Reply:
I think you should rename the expense repository to expense cache something. The name is not very explicit about what data source it is

Reply:
Okay, thinking about it on a higher level will definitely help. I'll rename it and remove the id returned. Thank you for the explanation